### PR TITLE
DAS-2063: Changes to use just LevelOrderIter

### PR DESCRIPTION
**Pull request by flamingbear**
Following this conversation, simplify the iterators.
https://github.com/pydata/xarray/pull/8879/files#r1557967100

### CODE REVIEW

**Comment by flamingbear** [comment]
Only extra thought was should it be renamed to something other than LevelOrderIter?

File: xarray/tests/test_datatree.py [hunk 1]
```diff
@@ -491,14 +491,15 @@ def test_nones(self):
     def test_full(self, simple_datatree):
         dt = simple_datatree
         paths = list(node.path for node in dt.subtree)
+        print(paths)
```
Comment (owenlittlejohns) [inline xarray/tests/test_datatree.py:494]: I was going to suggest removing `print`statement, but there are others throughout the file. Do you want it in here, or is it a debugging thing that can be removed?

Comment (flamingbear) [reply xarray/tests/test_datatree.py:494]: 😬 did I add that? Yes, it can get removed if so

Comment (flamingbear) [reply xarray/tests/test_datatree.py:494]: probably updating the tests

Comment (flamingbear) [reply xarray/tests/test_datatree.py]: updated

**Review by owenlittlejohns — APPROVED** [review]
Nice. Let's get this in and get this rolling. Thanks for getting this done so quickly.

**Comment by flamingbear** [comment]
Good stuff was plucked out of here. Closing this PR

**Pull request closed without merging**